From 33cc733abe19872bf5ed08540a94b798d0d4ecf4 Mon Sep 17 00:00:00 2001
From: Vito Gamberini <vito@gamberini.email>
Date: Wed, 7 Apr 2021 12:30:55 -0400
Subject: [PATCH] Remove unnecessary extern "C"s, fix build failures.

Wrapping libsigrockdecode.h in extern "C" is unecessary and will cause
build failures on modern versions of glib, this fixes those failures.
---
 DSView/pv/data/decode/annotation.cpp  | 2 --
 DSView/pv/view/decodetrace.cpp        | 2 --
 DSView/pv/widgets/decodergroupbox.cpp | 2 --
 3 files changed, 6 deletions(-)

diff --git a/DSView/pv/data/decode/annotation.cpp b/DSView/pv/data/decode/annotation.cpp
index c433a28..7f23e36 100755
--- a/DSView/pv/data/decode/annotation.cpp
+++ b/DSView/pv/data/decode/annotation.cpp
@@ -19,9 +19,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
  */
 
-extern "C" {
 #include <libsigrokdecode4DSL/libsigrokdecode.h>
-}
 
 #include <vector>
 #include <assert.h>
diff --git a/DSView/pv/view/decodetrace.cpp b/DSView/pv/view/decodetrace.cpp
index ad943ac..733497f 100755
--- a/DSView/pv/view/decodetrace.cpp
+++ b/DSView/pv/view/decodetrace.cpp
@@ -19,9 +19,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
  */
 
-extern "C" {
 #include <libsigrokdecode4DSL/libsigrokdecode.h>
-}
 
 #include <extdef.h>
 
diff --git a/DSView/pv/widgets/decodergroupbox.cpp b/DSView/pv/widgets/decodergroupbox.cpp
index 79d3cd8..80c4c44 100755
--- a/DSView/pv/widgets/decodergroupbox.cpp
+++ b/DSView/pv/widgets/decodergroupbox.cpp
@@ -18,9 +18,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
  */
-extern "C" {
 #include <libsigrokdecode4DSL/libsigrokdecode.h>
-}
 
 #include "decodergroupbox.h"
 #include "../data/decoderstack.h"

